lib/models/database/post_repository.dart

```diff
@@ -0,0 +1,278 @@
+import "package:cloud_firestore/cloud_firestore.dart";
+import "package:flutter/foundation.dart";
+import "package:geoflutterfire2/geoflutterfire2.dart";
+import "package:proxima/models/database/user_repository.dart";
+import "package:proxima/services/geolocation_service.dart";
+
+/// The id are strong typed to avoid misuse
+typedef PostFirestoreId = String;
+
+@immutable
+class PostFirestore {
+  /// The id is not stored in a field because it already
+  /// corresponds to the document id on firestore
+  final PostFirestoreId id;
+
+  /// The post location is not stored in the [PostFirestoreData] because it
+  /// is exclusively managed by the repository (in particular, it is the
+  /// responsability of the repository to create it when adding a post)
+  final PostLocationFirestore location;
+  static const String locationField = "location";
+
+  final PostFirestoreData data;
+
+  const PostFirestore({
+    required this.id,
+    required this.location,
+    required this.data,
+  });
+
+  /// Parses the data from a firestore document
+  factory PostFirestore.fromDb(DocumentSnapshot docSnap) {
+    if (!docSnap.exists) {
+      throw Exception("Post document does not exist");
```

Comment:
I agree on the difference between `Exception` and `Error`. However, I don't think that we should throw an error when the document doesn't exist because this something that could be expected (for instance, someone gets a post and at the same time someone else just deleted it). Thus, higher levels might want to catch it to display an error message. 

`FormatException` seems to be suitable for our usage (while not ideal as you said) so I will use them for parse exceptions, thanks for the suggestion !
The commit 1558e9f32d640d6b38f931739625caf8580c6613 add those.